### PR TITLE
added code to split the csv and host each nanocube

### PR DESCRIPTION
**Pull request by fdhoff**
$NANOCUBE_SRC and $NANOCUBE_BIN must be set up prior to running this Python script.

Now we need to pass the ports array to the distributed reader.

### CODE REVIEW

File: splitter.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#hostdmp convers the split csv files to dmp files, hosts them on ports 29512 and above, and removes the initial split .csv files
+def hostdmp(filename, numclusters):
+	ports = []
+	fn = filename.split('.csv')[0]+'_split'
+	for i in range(numclusters):
+		fileNum = i + 1
+		port = get_open_port()
+		ports.append(port)
+		dmp_command = "nanocube-binning-csv --sep=',' --timecol='time' --latcol='Latitude' --loncol='Longitude' \
+				--catcol='crime' " + filename.split('.csv')[0]+'_split' + str(fileNum) + '.csv > '\
```
Comment (asubiotto) [inline splitter.py]: You put fn on line 62 but didn't use it on line 68 and 69. Line 67 is also still hardcoded?

File: splitter.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+#hostdmp convers the split csv files to dmp files, hosts them on ports 29512 and above, and removes the initial split .csv files
+def hostdmp(filename, numclusters):
+	ports = []
+	fn = filename.split('.csv')[0]+'_split'
+	for i in range(numclusters):
+		fileNum = i + 1
+		port = get_open_port()
+		ports.append(port)
+		dmp_command = "nanocube-binning-csv --sep=',' --timecol='time' --latcol='Latitude' --loncol='Longitude' \
```
Comment (asubiotto) [inline splitter.py]: The column names are still being hardcoded here

File: splitter.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        port = s.getsockname()[1]
+        s.close()
+        return port
+
+
+#hostdmp convers the split csv files to dmp files, hosts them on ports 29512 and above, and removes the initial split .csv files
+def hostdmp(filename, numclusters):
+	ports = []
+	fn = filename.split('.csv')[0]+'_split'
+	for i in range(numclusters):
```
Comment (asubiotto) [inline splitter.py]: This should be multithreaded so the files are built in parallel. Start numCluster threads and then join them from the main thread to wait until their execution is done. [Here's an example](http://stackoverflow.com/questions/19138219/use-of-threading-thread-join) so the idea is to store a reference to each thread in a list and calling join on each thread in a loop at the end of this function.

**Comment by asubiotto** [comment]
I looked it up and lists in python are actually thread-safe (although their elements aren't of course) so we don't need a mutex for this. Your code looks good, but I'm still going to do a final check once we hand in ML and MS just to make sure. Just one comment about the submodules to take care of.

File: .gitmodules [hunk 4]
```diff
@@ -1,3 +0,0 @@
-[submodule "nanocube"]
```
Comment (asubiotto) [inline .gitmodules:1]: I think we forgot to move this file when we copied the directory into your GOPATH. Could you copy it over now?

**Comment by asubiotto** [comment]
@fdhoff took care of a few things:
- Removed the addition of extra time records
- Added the ability to run the script from anywhere (you just need to have $NANOCUBE_SRC set so it can find the binaries)
- Added a master port argument so that the user can specify which port to run the golang server on

Let me know what you think

**Comment by fdhoff** [comment]
Excellent

**Pull request merged**